embedding/cpp/flutter_service_app.cc

```diff
@@ -0,0 +1,173 @@
+// Copyright 2021 Samsung Electronics Co., Ltd. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "include/flutter_app.h"
+
+#include <service_app.h>
+
+#include <cassert>
+#include <cerrno>
+#include <fstream>
+
+#include "tizen_log.h"
+
+bool FlutterApp::OnCreate() {
+  TizenLog::Debug("Launching a Flutter Service application...");
+
+  std::string res_path;
+  {
+    auto path = app_get_resource_path();
+    if (path == nullptr) {
+      TizenLog::Error("Could not obtain the app directory info.");
+      return false;
+    }
+    res_path = path;
+    free(path);
+  }
+  std::string assets_path(res_path + "/flutter_assets");
+  std::string icu_data_path(res_path + "/icudtl.dat");
+  std::string aot_lib_path(res_path + "/../lib/libapp.so");
+
+  // Read engine arguments passed from the tool.
+  ParseEngineArgs();
+
+  std::vector<const char *> switches;
+  for (auto &arg : engine_args) {
+    switches.push_back(arg.c_str());
+  }
+
+  FlutterEngineProperties engine_prop = {};
+  engine_prop.assets_path = assets_path.c_str();
+  engine_prop.icu_data_path = icu_data_path.c_str();
+  engine_prop.aot_library_path = aot_lib_path.c_str();
+  engine_prop.switches = switches.data();
+  engine_prop.switches_count = switches.size();
+
+  handle = FlutterRunEngine(engine_prop);
+  if (!handle) {
+    TizenLog::Error("Could not launch a Flutter Service application.");
+    return false;
+  }
+
+  return true;
+}
+
+void FlutterApp::OnResume() {
```

Comment:
most of this file is a duplication of flutter_app.cc maybe we can consider using inheritance to reduce redundancy of code. Is it possible or there are some contraindications?

Reply:
I tried to make everything as simple as possible. But I will check if there is any options of reducing code.